catroid/src/main/java/org/catrobat/catroid/ui/dialogs/LegoSensorPortConfigDialog.java

```diff
@@ -28,160 +28,112 @@
-					title = getString(R.string.nxt_sensor_sound);
-					nxtSensor = NXTSensor.Sensor.SOUND;
-					break;
-				case R.string.formula_editor_sensor_lego_nxt_light:
-					title = getString(R.string.nxt_sensor_light);
-					nxtSensor = NXTSensor.Sensor.LIGHT_INACTIVE;
-					break;
-				case R.string.formula_editor_sensor_lego_nxt_light_active:
-					title = getString(R.string.nxt_sensor_light_active);
-					nxtSensor = NXTSensor.Sensor.LIGHT_ACTIVE;
-					break;
-				case R.string.formula_editor_sensor_lego_nxt_ultrasonic:
-					title = getString(R.string.nxt_sensor_ultrasonic);
-					nxtSensor = NXTSensor.Sensor.ULTRASONIC;
-					break;
-				default:
-					title = getString(R.string.nxt_sensor_not_found);
-					nxtSensor = NXTSensor.Sensor.NO_SENSOR;
-			}
-		} else {
-			switch (clickedItem) {
-				case R.string.formula_editor_sensor_lego_ev3_sensor_touch:
-					title = getString(R.string.ev3_sensor_touch);
-					ev3Sensor = EV3Sensor.Sensor.TOUCH;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_infrared:
-					title = getString(R.string.ev3_sensor_infrared);
-					ev3Sensor = EV3Sensor.Sensor.INFRARED;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color:
-					title = getString(R.string.ev3_sensor_color);
-					ev3Sensor = EV3Sensor.Sensor.COLOR;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color_ambient:
-					title = getString(R.string.ev3_sensor_color_ambient);
-					ev3Sensor = EV3Sensor.Sensor.COLOR_AMBIENT;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_color_reflected:
-					title = getString(R.string.ev3_sensor_color_reflected);
-					ev3Sensor = EV3Sensor.Sensor.COLOR_REFLECT;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_hitechnic_color:
-					title = getString(R.string.ev3_sensor_hitechnic_color);
-					ev3Sensor = EV3Sensor.Sensor.HT_NXT_COLOR;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_nxt_temperature_c:
-					title = getString(R.string.ev3_sensor_nxt_temperature_c);
-					ev3Sensor = EV3Sensor.Sensor.NXT_TEMPERATURE_C;
-					break;
-				case R.string.formula_editor_sensor_lego_ev3_sensor_nxt_temperature_f:
-					title = getString(R.string.ev3_sensor_nxt_temperature_f);
-					ev3Sensor = EV3Sensor.Sensor.NXT_TEMPERATURE_F;
-					break;
-				default:
-					title = getString(R.string.nxt_sensor_not_found);
-					ev3Sensor = EV3Sensor.Sensor.NO_SENSOR;
-			}
-		}
+	private SensorInfo getSensorInfo(int clickedItem, Lego type) {
+		SensorInfo info = sensorInfoMap.get(clickedItem);
+		Enum sensor = type == Lego.NXT ? NXTSensor.Sensor.NO_SENSOR : EV3Sensor.Sensor.NO_SENSOR;
+		return info != null ? info : new SensorInfo(getString(R.string.nxt_sensor_not_found), sensor);
 	}
 
 	private void checkIfOverwrite(final int selectedPort, String selected, final Lego legoType) {
-		final Context dialogueContext = getActivity();
+		final Context dialogContext = getActivity();
```

Comment:
Please remove the local variable -> use getActivty() directly in the AlertDialog.Builder constructor.